hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -1551,4 +1552,30 @@ private static Pair<Boolean, String> getIsValueWrapperObfuscated(GenericRecord r
     }
     return Pair.of(false, null);
   }
+
+  /**
+   * Returns field name and the resp data type of the field. The data type will always refer to the leaf node.
+   * for eg, for a.b.c, we turn Pair.of(a.b.c, DataType(c))
+   * @param schema schema of table.
+   * @param fieldName field name of interest.
+   * @return
+   */
+  @VisibleForTesting
+  public static Pair<String, Schema.Field> getSchemaForField(Schema schema, String fieldName) {
+    return getSchemaForField(schema, fieldName, StringUtils.EMPTY_STRING);
+  }
+
+  @VisibleForTesting
+  public static Pair<String, Schema.Field> getSchemaForField(Schema schema, String fieldName, String prefix) {
```

Comment:
Can we please add direct UTs for the methods with simple field, nested field, nullable/non-nullable? I know it's already being tested for nested fields in TestColumnStatsIndex. But, it would be better to cover all cases including edge cases in a separate UT for this methd itself.